### PR TITLE
updates to the Trigger code + small fix in CaloHitMakerFast

### PR DESCRIPTION
**Pull request by gianipez**
- improvements to the Trigger Analyzer module
- now setting the paths ID in the Menu files
- update the TriggerResultsNavigator code to handle the labels of the `art::TriggerResults` object of the latest art version ("pathName" -> "id:pathName")
- temporary fix in the `CaloHitMakerFast` module (see comments in the code)
- fixed time offset for CalPatRec-trigger_mode (probably needs to be fixed also for the Offline_mode); it moved from `-41` to `-61`
- probably need to adjust the calorimeter-timeOffset also for TPR; I tried to apply the same correction I found for CPR, but it didn't play as expected (aka smaller eff); right now is `-20`, and I tried `-40` and `0`, but both tests were returning worst result than the default.

### CODE REVIEW

**Comment by kutschke** [comment]
@FNALbuild run build tests

File: CaloReco/src/CaloHitMakerFast_module.cc
Comment (kutschke) [inline CaloReco/src/CaloHitMakerFast_module.cc:114]: Again for future reference.  I like that you are using .at(i) instead of [i] for development.  For time critical code we should switch it back to [i] for production.  I know of several ways to way to write code so that we can flip between the two with a compiler switch.  When you are ready let's talk about it.

File: Mu2eUtilities/src/TriggerResultsNavigator.cc
Comment (kutschke) [inline Mu2eUtilities/src/TriggerResultsNavigator.cc:61]: How will this function be used?  I generally don't like returning a magic string to denote an error.  But there are probably use cases in which it is the best way - for example if it's only being used in printout and it's normal that it might sometimes be missing.

File: Trigger/python/genTriggerFcl.py
Comment (kutschke) [inline Trigger/python/genTriggerFcl.py:243]: How many other pieces of code need to know that 100 is a magic number?   I see that this is in the python world.  Will we need a place in the C++ world or the tuple world?  How can we define it once in each world?

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [inline Trigger/src/ReadTriggerInfo_module.cc:298]: We are asking everyone to convert modules to validated fcl.  Please do so.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [inline Trigger/src/ReadTriggerInfo_module.cc:316]: Please  make a sub-table to hold the cuts.  When you make the change, remove the defaults from the code.

**Review by kutschke — CHANGES_REQUESTED** [review]
I have a few structural change requests and a few suggestions for future improvements.  

Do you expect this to change validation?  As best I can tell the change to dtOffset in CalPatRec/fcl/prolog_trigger.fcl is not seen by validation.   And I don't think that any of the modules are seen by validation.

File: CalPatRec/fcl/prolog_trigger.fcl
Comment (gianipez) [reply CalPatRec/fcl/prolog_trigger.fcl:130]: I think that once that this offset could be included in the calorimeter channel time calibration, which will differ for Offline and Online reconstruction. Right now, there is no such a prodiction.

File: CaloReco/src/CaloHitMakerFast_module.cc
Comment (gianipez) [reply CaloReco/src/CaloHitMakerFast_module.cc:114]: Interesting. If this is something that affects several parts of the Offline it might be good to identify someone who has time to make this change.

File: Mu2eUtilities/src/TriggerResultsNavigator.cc
Comment (gianipez) [reply Mu2eUtilities/src/TriggerResultsNavigator.cc:61]: I honestly wasn't happy either, but I thought that there might be analysis code that could be ok with having a pathname not found.

File: Trigger/python/genTriggerFcl.py
Comment (gianipez) [reply Trigger/python/genTriggerFcl.py:243]: That is something that was discussed a long time ago in some software meeting; I proposed to leave free the first 100 bits for non-triggger paths. No other code needs to know this number. This script generates only the trigger Menu file. An analyzer will need to know either the list of the "good" bits or the list of the pathnames.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (gianipez) [reply Trigger/src/ReadTriggerInfo_module.cc:316]: I can do it, but I'm not super excited about having a table for an analyzer module in the prolog_trigger.fcl file. Are we sure about it?

File: CalPatRec/fcl/prolog_trigger.fcl
Comment (kutschke) [reply CalPatRec/fcl/prolog_trigger.fcl:130]: THanks.  Just keep it on your radar when the proditions object does exist.

File: CaloReco/src/CaloHitMakerFast_module.cc
Comment (kutschke) [reply CaloReco/src/CaloHitMakerFast_module.cc:114]: Do you have any summer students who could do it?

File: Mu2eUtilities/src/TriggerResultsNavigator.cc
Comment (kutschke) [reply Mu2eUtilities/src/TriggerResultsNavigator.cc:61]: The standard answer is to define a small handle-like class and return that.  It would have two data members, the string and bool isValid.  The data should be private and it should throw if you someone tries to access an invalid string.

I have asked Marc and Kyle to see if a standard one exists.  If not I can write one as a template and you can use it.  No need to hold up this PR.

File: CaloReco/src/CaloHitMakerFast_module.cc
Comment (gianipez) [reply CaloReco/src/CaloHitMakerFast_module.cc:114]: No. The summer students working with my group have other projects assigned on the trigger front.

Comment (kutschke) [reply CaloReco/src/CaloHitMakerFast_module.cc:114]: I created an issue to followup on this general. 

If we do this in a few places the time cost is small.  If it's ubiquitous then it can add up to a large time cost that is invisible to profilers because it's distributed throughout the code.

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Comment by kutschke** [comment]
@FNALbuild run build tests

Expect ceMix to fail until the input files are updated.

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Review by kutschke — APPROVED** [review]
This got lost in the hectic time of the collaboration meeting.  Approving now.  As you move forward please address the issues that I raised.

**Pull request merged**
